plugins/buildClusters/update.js

```diff
@@ -0,0 +1,125 @@
+
+                if (!adminDetails.isAdmin) {
+                    return reply(boom.forbidden(
+                        `User ${adminDetails.userDisplayName}
+                        does not have Screwdriver administrative privileges.`
+                    ));
+                }
+
+                return buildClusterFactory.list({
+                    params: {
+                        name
+                    }
+                })
+                    .then((buildCluster) => {
+                        if (!buildCluster) {
+                            throw boom.notFound(`Build cluster ${name} does not exist`);
+                        }
+
+                        Object.assign(buildCluster, request.payload);
+
+                        return buildCluster.update()
+                            .then(updatedBuildCluster =>
+                                reply(updatedBuildCluster.toJson()).code(200)
+                            );
+                    })
+                    .catch(err => reply(boom.boomify(err)));
+            }
+
+            // Must have admin permission on org if adding org-specific build cluster
+            if (scmOrganizations && scmOrganizations.length === 0) {
+                return reply(boom.boomify(boom.badData(
+                    `No scmOrganizations provided for build cluster ${name}.`
+                )));
+            }
+
+            return userFactory.get({ username, scmContext })
+                .then(user => Promise.all([
+                    user.unsealToken(),
+                    buildClusterFactory.list({
+                        params: {
+                            name
+                        }
+                    })])
+                    .then(([token, buildCluster]) => {
+                        if (!buildCluster) {
+                            throw boom.notFound(`Build cluster ${name} does not exist`);
+                        }
+
+                        // To update scmOrganizations, user need to have admin permissions on both old and new organizations
+                        const orgs = buildCluster.scmOrganizations;
+                        const newOrgs = scmOrganizations || [];
+                        const combined = [...new Set([...orgs, ...newOrgs])];
+
+                        return Promise.all(combined.map(organization =>
+                            scm.getOrgPermissions({
+                                organization,
+                                username,
+                                token,
+                                scmContext
+                            })
+                                .then((permissions) => {
+                                    if (!permissions.admin) {
+                                        throw boom.forbidden(
+                                            `User ${username} does not have
+                                            administrative privileges on build
+                                            cluster ${name}.`
```

Comment:
should this be
```
                                            `User ${username} does not have
                                            administrative privileges on scm 
                                            organization ${organization}.`
```
instead of on the build cluster?